crates/ty_python_semantic/src/types/class.rs

```diff
@@ -2138,6 +2157,13 @@ impl<'db> ClassLiteral<'db> {
             member = member.map_type(|ty| into_function_like_callable(db, ty));
         }
 
+        // For callable members that use `Self` type variables (e.g., synthesized methods like
+        // `_replace` on NamedTuples), replace `Self` with the instance type of this class.
+        // This ensures that when accessing `Person._replace`, we get `(self: Person, ...) -> Person`
+        // instead of `(self: Self, ...) -> Self`.
+        let self_instance = Type::instance(db, ClassType::NonGeneric(self));
+        member = member.map_type(|ty| apply_self_to_callable(db, ty, self_instance));
```

Comment:
What's not clear to me is why we are accessing `person._replace` on the class to begin with, when we are calling it on an instance. That's something we do for dunder methods (to mimic the runtime, which also does that), but it's not something we should do for a normal instance method call; `_replace` is not a dunder method. That seems to be at the root of the need for this.